server/client.go

```diff
@@ -452,6 +455,15 @@ func (c *client) processConnect(arg []byte) error {
 			c.authViolation()
 			return ErrAuthorization
 		}
+
+		// Client-enabled tracing. Check permissions if applicable.
+		if c.opts.Trace {
+			if srv.opts.OptInTrace || (c.perms != nil && c.perms.trace) {
+				c.trace = true
+			} else {
+				c.sendErr("Permissions Violation for trace option")
```

Comment:
Since an error is sent back, the client will close the connection. So you need to recreate a client and do the connect again for the following success case scenario.